src/Featurizers/AnalyticalRollingWindowFeaturizer.h

```diff
@@ -0,0 +1,305 @@
+///                 so we are peparing by making this an enum from the get go.
+///
+enum class AnalyticalRollingWindowCalculation : std::uint8_t {
+    Mean = 1
+};
+
+
+/////////////////////////////////////////////////////////////////////////
+///  \class         AnalyticalRollingWindowTransformer
+///  \brief         RollingWindow class that is used for all analytical window computations.
+///
+template <
+    typename InputT,
+    size_t MaxNumTrainingItemsV=std::numeric_limits<size_t>::max()
+>
+class AnalyticalRollingWindowTransformer:
+    public Components::InferenceOnlyTransformerImpl<InputT, std::vector<double>> {
+public:
+    // ----------------------------------------------------------------------
+    // |
+    // |  Public Types
+    // |
+    // ----------------------------------------------------------------------
+    using BaseType = Components::InferenceOnlyTransformerImpl<InputT, std::vector<double>>;
+
+    // ----------------------------------------------------------------------
+    // |
+    // |  Public Methods
+    // |
+    // ----------------------------------------------------------------------
+    AnalyticalRollingWindowTransformer(std::uint32_t windowSize, AnalyticalRollingWindowCalculation windowCalculation, std::uint32_t horizon, std::uint32_t minWindowCount = 1) ;
+    AnalyticalRollingWindowTransformer(Archive &ar);
+    ~AnalyticalRollingWindowTransformer(void) override = default;
+
+    FEATURIZER_MOVE_CONSTRUCTOR_ONLY(AnalyticalRollingWindowTransformer);
+
+    bool operator==(AnalyticalRollingWindowTransformer const &other) const;
+    bool operator!=(AnalyticalRollingWindowTransformer const &other) const;
+    void save(Archive & ar) const override;
+
+private:
+    // ----------------------------------------------------------------------
+    // |
+    // |  Private Members
+    // |
+    // ----------------------------------------------------------------------
+    // _windowSizes and _minWindowCount are int64 so that we can have the full positive range of the uint32 it comes
+    // in as but can still support negative numbers.
+    Components::CircularBuffer<InputT>              _buffer;
+    const std::uint32_t                             _windowSize;
+    const std::uint32_t                             _horizon;
+    const std::uint32_t                             _minWindowSize;
+    const AnalyticalRollingWindowCalculation        _windowCalculation;
+
+    // ----------------------------------------------------------------------
+    // |
+    // |  Public Methods
+    // |
+    // ----------------------------------------------------------------------
+
+    // MSVC runs into problems when the declaration and definition are separated
+    void execute_impl(typename BaseType::InputType const &input, typename BaseType::CallbackFunction const &callback) override {
+        typename BaseType::TransformedType         results(_horizon);

```

Comment:
>typename BaseType::TransformedType         results(_horizon); [](start = 8, length = 61)

Declare this closer to the code where it is used.

Reply:
Moved it down to right above the for loop.